### ISSUE TITLE
Mozilla user should be special and no-one should be able to log in as it

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#8735]
The Mozilla user (https://addons.mozilla.org/firefox/user/mozilla/) owns a whole bunch of high profile add-ons and collections.  We shouldn't allow anyone to log in as it - even if they have a valid FxA login for the linked email address (addons-team@m.c).

(Plus we special case some actions & logging attributed to it because it's also used for some tasks like auto approval)

**Comment by eviljeff** [comment]
Upping priority because it's a hypothetical risk we should fix sooner rather than later.

**Comment by AlexandraMoga** [comment]
Following @eviljeff 's recommendations, I followed the below steps in order to test this feature:

1. Edited the mozilla account in `/admin.models/users/`
2. Replaced the addons-team@m.c email with a personal email address (the address belonged to one of my accounts that was deleted in the meantime)
3. Made sure the FxA id field was empty
4. Tried to log in to AMO after the changes were applied

Result:
I received the following message from the authentication api:
![image](https://user-images.githubusercontent.com/31961530/45352418-1d518f00-b5c1-11e8-9b6b-622d5c7db6fb.png)

@eviljeff does this look alright?
